testbeam_analysis/gui/analysis_threading.py

```diff
@@ -0,0 +1,101 @@
+import yaml
+import traceback
+
+from multiprocessing import Pool
+from PyQt5 import QtCore
+
+
+class AnalysisThread(QtCore.QThread):
+    """
+    Implements a class which allows to perform analysis / start vitables on an
+    extra thread to keep the GUI responsive during analysis / vitables
+    """
+
+    exceptionSignal = QtCore.pyqtSignal(Exception)
+
+    def __init__(self, func, args=None, funcs_args=None, parent=None):
+
+        super(AnalysisThread, self).__init__(parent)
+
+        # Main function which will be executed on this thread
+        self.main_func = func
+        # Arguments of main function
+        self.args = args
+        # Functions and arguments to perform analysis function;
+        # if not None, main function is then AnalysisWidget.call_funcs()
+        self.funcs_args = funcs_args
+
+    def run(self):
```

Comment:
That is subclassing the run method of QThread and not the best way. I guess this is not needed anymore and can be deleted?